x-pack/plugins/maps/public/index.ts

```diff
@@ -10,3 +10,5 @@ import { MapsPlugin, MapsPluginSetup, MapsPluginStart } from './plugin';
 export const plugin: PluginInitializer<MapsPluginSetup, MapsPluginStart> = () => {
   return new MapsPlugin();
 };
+
+export { MAP_SAVED_OBJECT_TYPE } from '../common/constants';
```

Comment:
Not sure this export should be added to this PR. There are lots of places in `kibana/x-pack/legacy/plugins` that still import `MAP_SAVED_OBJECT_TYPE` from `/common/constants` and I would rather update all imports so they are consistent vs having inconsistent imports. The scope of this PR is starting to get rather large so maybe just save this for later?

Reply:
SIEM was deep importing which is not allowed, but getting around it by adding the @ts-ignore.  I think inconsistent imports are an improvement over invalid deep linking and using @ts-ignore which will hide any errors.

I think it's fine if internally users import from common directly, but externally, you are only supposed to import from server or public top level. (Though support is in the works for common exports - https://github.com/elastic/kibana/issues/59710)

Reply:
The exports were from external users like canvas, siem, and update. Maybe update all those locations so they are consist then?

https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/canvas/canvas_plugin_src/expression_types/embeddable_types.ts#L8

https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/uptime/public/components/functional/location_map/embeddables/embedded_map.tsx#L14

https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/siem/public/components/embeddables/embedded_map_helpers.tsx#L15